src/lib/sys.c

```diff
@@ -348,3 +366,86 @@ bool is_dir(const char *path)
 
 	return true;
 }
+
+static int sys_rm_file(const char *path)
+{
+	if (unlink(path) != 0) {
+		return -errno;
+	}
+
+	return 0;
+}
+
+static int sys_rmdir(const char *path)
+{
+	if (rmdir(path) != 0) {
+		return -errno;
+	}
+
+	return 0;
+}
+
+static int sys_rm_dir_recursive(const char *path)
```

Comment:
I see the recursiveness will take care of other directories within the directory, but what if any of the directories still have files in it?

Reply:
we call sys_rm_recursive() that will remove files too.